File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/dto/response/SubmissionResponseDTO.kt
package io.leonfoliveira.judge.api.dto.response

import io.leonfoliveira.judge.core.domain.entity.Submission

data class SubmissionResponseDTO(
    val id: Int,
    val problemId: Int,
    val memberId: Int,
    val status: String,
    val language: String,
    val createdAt: String,
    val updatedAt: String,
)

fun Submission.toResponseDTO(): SubmissionResponseDTO {
    return SubmissionResponseDTO(
        id = id,
        problemId = problem.id,
        memberId = member.id,
        status = status.name,
        language = language.name,
        createdAt = createdAt.toString(),
        updatedAt = updatedAt.toString(),
    )
}

File: judge-web/src/core/repository/dto/response/SubmissionResponseDTO.ts
import { SubmissionStatus } from "@/core/domain/enumerate/SubmissionStatus";
import { Language } from "@/core/domain/enumerate/Language";
import { DownloadAttachmentResponseDTO } from "@/core/repository/dto/response/DownloadAttachmentResponseDTO";

export type SubmissionResponseDTO = {
  id: number;
  problem: {
    id: number;
    title: string;
  };
  member: {
    id: number;
    name: string;
  };
  language: Language;
  status: SubmissionStatus;
  code: DownloadAttachmentResponseDTO;
  createdAt: string;
  updatedAt: string;
};
